core/validation/failures/user.py

```diff
@@ -295,10 +295,11 @@ def failures_to_messages(
 ) -> dict[str, list[str]] | dict[str, list[dict[str, str | None]]]:
     """Serialises failures into messages, removing any duplicates, and groups them by tab and section.
     :param validation_failures: validation failure objects
-    :return: validation failure messages grouped by tab and section
+    :return: validation failure messages grouped by tab and section and list of error types
     """
     # filter and convert to error messages
     error_messages = [failure.to_message() for failure in validation_failures]
```

Comment:
I think a an improvement could be to add the `class` to the message tuple instead, like

```
error_messages = [(*failure.to_message(), failure.__class__.__name__) for failure in validation_failures]
```

this way we keep all the information for a single failure together and minimise the changes to the return structure of `/ingest`

Reply:
I didn't see @Connor-OS's comment before I added this. We're alluding to the same point